metamask-with/metamask-sdk-nextjs-pages-directory/components/SdkProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import { PropsWithChildren, useEffect } from 'react'
+import { useListen } from '../hooks/useListen'
+import { useMetaMask } from '../hooks/useMetaMask'
+import { instantiateSdk } from '../lib/metamaskSDK'
+
+export const SdkLayout: React.FC<PropsWithChildren> = ({ children }) => {
+  const { dispatch } = useMetaMask()
+  const listen = useListen()
+
+  useEffect(() => {
+    if (typeof window !== undefined) {
+      // start by checking if window.ethereum is present, indicating a wallet extension
```

Comment:
Fully agree with more white spaces to help readability. Not sure however between 12-13 since the comment is meant for the line right under. I added one between 13-14.

I also can't add a line after the if statement as [Prettier removes those automatically](https://prettier.io/docs/en/rationale.html#empty-lines). I haven't seen the option to change this while searching.